src/msgpack/Converters/BinaryConverter.cs

```diff
@@ -4,7 +4,7 @@ namespace TarantoolDnx.MsgPack.Converters
 {
     internal class BinaryConverter : IMsgPackConverter<byte[]>
     {
-        public void Write(byte[] value, IMsgPackWriter writer, MsgPackContext context)
+        public void Write(byte[] value, IBytesWriter writer, MsgPackContext context)
```

Comment:
Imho, IMsgPackWriter is not perfect name. But I agree, it is much better that IBytesReader.
Renamed, IBytesWriter also. (and implementation classes).